### PR TITLE
changed jl_error to Throw ArgumentError

### LINKED ISSUE #32307: Array allocation with negative size should throw a `ArgumentError` not a `ErrorException`
```
julia> try 
       fill('a', -10)
       catch err
       @show typeof(err)
       end

typeof(err) = ErrorException
```

### PR DESCRIPTION
**Pull request by dnabanita7**
Fix #32307 
Added a throw ArgumentError

### CODE REVIEW

File: src/array.c [hunk 1]
```diff
@@ -64,7 +64,7 @@ static jl_array_t *_new_array_(jl_value_t *atype, uint32_t ndims, size_t *dims,
         size_t di = dims[i];
         wideint_t prod = (wideint_t)nel * (wideint_t)di;
         if (prod > (wideint_t) MAXINTVAL || di > MAXINTVAL)
-            jl_error("invalid Array dimensions");
+            throw ArgumentError("invalid Array dimensions");
```
Comment (StefanKarpinski) [inline src/array.c]: This file is C code, not Julia code. The changed code here is not valid C.

**Comment by mbauman** [comment]
I recommend [building Julia](https://github.com/JuliaLang/julia#building-julia) to test your change — and that will help you come up with test cases to add to the tests like Kristoffer suggested.  It'll also ensure that you have your syntax correct — this doesn't build currently as it's not valid C code.  You want to do something like I suggested at the end of my comment here: https://github.com/JuliaLang/julia/pull/32341#issuecomment-502711019

File: src/array.c (hunk 1 above)
Comment (KristofferC) [reply src/array.c]: This is also not valid Julia code.

**Comment by Naba7** [comment]
I get an error while building Julia: 

```
curl: (28) Resolving timed out after 15542 milliseconds
/home/chad7/julia/deps/suitesparse.mk:26: recipe for target '/home/chad7/julia/deps/srccache/SuiteSparse-4.4.5.tar.gz' failed
make[1]: *** [/home/chad7/julia/deps/srccache/SuiteSparse-4.4.5.tar.gz] Error 28
Makefile:57: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2
```

**Comment by StefanKarpinski** [comment]
Looks like you might be behind a fire wall?

**Comment by Naba7** [comment]
I used ``make USE_SYSTEM_...=1`` and its working now but giving warnings in openblas as:
https://gist.github.com/Naba7/5776b18768691eb5bc09ca3d48ec2c28

**Comment by Naba7** [comment]
after building julia from source again, I have created another pr for this. Thanks for the help.

**Pull request closed without merging**